### PR TITLE
4204 - Update email templates

### PR DESCRIPTION
**Pull request by avinhurry**
## Context

Updates to the content on two of our email templates, this includes the addition of logic to render different text, dependendent on conditional statements.

## Changes proposed in this pull request

- Update `application_submitted` email template
- Update `application_submitted_apply_again` email template
- Add dynamic content to both of these templates

### Application Submitted (Before)

image

### Application Submitted (After)
image

### Application Submitted (multiple courses)(After)

image

### Application Submitted Apply Again (Before)

image

### Application Submitted Apply Again(After)

image

## Guidance to review

- Could `application_form.application_choices` ever be `nil` at the point that this email is fired? 🤔 
- Could `reject_by_default_at`ever be `nil` at the point that this email is fired? 🤔 

- You can view the new Application Submitted email [here](https://apply-for-te-4204-chang-17flie.herokuapp.com/rails/mailers/candidate_mailer/application_submitted).
- You can view the new Application Submitted Apply Again [here](https://apply-for-te-4204-chang-17flie.herokuapp.com/rails/mailers/candidate_mailer/application_submitted_apply_again).

### CODE REVIEW

**Comment by avinhurry** [comment]
[Changes to post submission emails (Apply 1 and 2)](https://trello.com/c/nxMKF73p/4204-changes-to-post-submission-emails-apply-1-and-2)

File: app/views/candidate_mailer/application_submitted.text.erb [hunk 1]
```diff
@@ -1,13 +1,23 @@
...
-You have submitted an application for:
+You have submitted an application for <%= @application_choice.course_option.course.name_and_code %> at <%= @application_choice.course_option.course.provider.name %>.
+
+Sign in to your account to check the progress of your application:
+
+<%= @candidate_magic_link %>
+
+<% if @application_form.application_choices.count > 1 %>
+
+  If your training providers decide to progress your application, they’ll contact you to organise an interview.
```
Comment (EmmaFrith) [inline app/views/candidate_mailer/application_submitted.text.erb:11]: if the application choices count is one, shouldn't this first line be the singular, and the line after else the plural? @avinhurry

File: app/views/candidate_mailer/application_submitted.text.erb [hunk 2]
```diff
@@ -1,13 +1,23 @@
 Dear <%= @application_form.first_name %>,
```
Comment (EmmaFrith) [inline app/views/candidate_mailer/application_submitted.text.erb:1]:
```suggestion
Dear <%= @application_form.first_name %>
```

File: app/views/candidate_mailer/application_submitted_apply_again.text.erb [hunk 3]
```diff
@@ -1,11 +1,23 @@
 Dear <%= @application_form.first_name %>,
```
Comment (EmmaFrith) [inline app/views/candidate_mailer/application_submitted_apply_again.text.erb:1]:
```suggestion
Dear <%= @application_form.first_name %>
```

File: app/views/candidate_mailer/application_submitted.text.erb (hunk 1 above)
Comment (avinhurry) [reply app/views/candidate_mailer/application_submitted.text.erb:11]: It's if the count is greater than `1 (>)`, so if it is greater than 1 it would output the line of text below it. 

So that means we default to the singular for anything else, ie (potential bugs where the count ends up being `nil`) which kind of made more sense to me, but happy to change it around if you think we should. Your suggestion is probably more readable.

File: app/views/candidate_mailer/application_submitted.text.erb (hunk 2 above)
Comment (avinhurry) [reply app/views/candidate_mailer/application_submitted.text.erb:1]: Thanks. Good 👀 , will fix!

File: app/views/candidate_mailer/application_submitted_apply_again.text.erb (hunk 3 above)
Comment (avinhurry) [reply app/views/candidate_mailer/application_submitted_apply_again.text.erb:1]: Same here 😄

File: app/views/candidate_mailer/application_submitted.text.erb (hunk 1 above)
Comment (EmmaFrith) [reply app/views/candidate_mailer/application_submitted.text.erb:11]: ahhhh didn't see the greater than sign - thanks Avin!

Comment (frankieroberto) [reply app/views/candidate_mailer/application_submitted.text.erb:11]: If they've submitted the application with more than one choice, then the first line ("You have submitted an application for X at Y") doesn't make sense?

File: app/views/candidate_mailer/application_submitted_apply_again.text.erb [hunk 4]
```diff
@@ -1,11 +1,23 @@
...
-If your training provider decides to progress your application, they’ll contact you to organise an interview.
+<% if @application_form.application_choices.count > 1 %>
+
+  If your training providers decide to progress your application, they’ll contact you to organise an interview.
+
+<% else %>
+
+  If your training provider decides to progress your application, they’ll contact you to organise an interview.
+
+<% end %>
```
Comment (frankieroberto) [inline app/views/candidate_mailer/application_submitted_apply_again.text.erb:17]: We shouldn't need this condition, as you can only have 1 choice in Apply again?

Comment (EmmaFrith) [reply app/views/candidate_mailer/application_submitted_apply_again.text.erb:17]: good point! (although, we might be changing business rules so that you can apply for more than one in the not too distant future)

File: app/views/candidate_mailer/application_submitted.text.erb (hunk 1 above)
Comment (EmmaFrith) [reply app/views/candidate_mailer/application_submitted.text.erb:11]: when it's more than one I think the line reads: 

You have submitted an application for: [list course names and provider names]

(so it's 1 application, for multiple courses/providers)

We're not consistent with application/applications in the service - something to look at in the future

Comment (frankieroberto) [reply app/views/candidate_mailer/application_submitted.text.erb:11]: @EmmaFrith I’m not sure the code in this PR right now does that - the bullet list has been removed and line 3 just refers to a single course? Unless I missing something... @avinhurry ?

**Review by EmmaFrith — APPROVED** [review]

File: app/views/candidate_mailer/application_submitted.text.erb (hunk 1 above)
Comment (avinhurry) [reply app/views/candidate_mailer/application_submitted.text.erb:11]: Yeah good spot @frankieroberto, you are right. `You have submitted an application for...` (singular) isn't accurate if they have submitted more than one application. 

Do we need to make that line conditional too? Can we update the doc with what we want it to be please. https://docs.google.com/document/d/1v8eGxJuWU0JGoP71zT2sgFuBRpqO80V4-oHpTaYi6f8/edit

File: app/views/candidate_mailer/application_submitted_apply_again.text.erb (hunk 4 above)
Comment (avinhurry) [reply app/views/candidate_mailer/application_submitted_apply_again.text.erb:17]: ooh thats a good point, that slipped my mind. Shall we remove it or leave it in for future proofing? 🤔

File: app/views/candidate_mailer/application_submitted.text.erb (hunk 1 above)
Comment (EmmaFrith) [reply app/views/candidate_mailer/application_submitted.text.erb:11]: You have submitted an application for the following courses:

[list courses and providers applied for]

could work 


---

We could change it to 'applications' if it's plural but if we do that we should do that throughout the entire email. We use 'application' quite a lot. E.g. one application for multiple courses.

Comment (frankieroberto) [reply app/views/candidate_mailer/application_submitted.text.erb:11]: @avinhurry I don't just mean the pluralisation - doesn't line 3 currently only list a single course?

```ruby
You have submitted an application for <%= @application_choice.course_option.course.name_and_code %> at <%= @application_choice.course_option.course.provider.name %>.
```

?

Comment (avinhurry) [reply app/views/candidate_mailer/application_submitted.text.erb:11]: Discussed offline - the decision is to leave line 3 in for singular and render the loop which was previously here if its plural.

File: app/views/candidate_mailer/application_submitted_apply_again.text.erb (hunk 4 above)
Comment (avinhurry) [reply app/views/candidate_mailer/application_submitted_apply_again.text.erb:17]: Discussed offline - the decission is to remove the conditional on line 9 as currently you can only have one choice in Apply again

File: spec/system/support_interface/see_individual_application_spec.rb [hunk 5]
```diff
@@ -35,6 +35,9 @@ def and_there_are_applications_in_the_system
       :with_gcses,
       references_state: :feedback_requested,
     )
+
+    create(:application_choice, application_form: @completed_application, status: 'unsubmitted')
```
Comment (stevehook) [inline spec/system/support_interface/see_individual_application_spec.rb:39]: why is this needed?

**Review by stevehook — COMMENTED** [review]
Looks good, I had a few minor question inline...

File: spec/system/support_interface/see_individual_application_spec.rb (hunk 5 above)
Comment (avinhurry) [reply spec/system/support_interface/see_individual_application_spec.rb:39]: I think we need it to create an instance of `application_choice` so that `application_form.application_choices.first` isn't `nil` so the mailer can call `reject_by_default_at` on it

File: spec/system/support_interface/email_log_spec.rb [hunk 6]
```diff
@@ -37,9 +38,22 @@ def when_an_email_with_custom_reference_is_sent
...
+      first_name: 'Harry',
+      last_name: 'Potter',
+      candidate: @candidate,
+    )
+
+    create(:application_choice, application_form: @completed_application, status: 'unsubmitted')
+
+    SubmitApplication.new(@completed_application).call
+  end
+
```
[the remark this answers is not in the document]
Comment (avinhurry) [reply spec/system/support_interface/email_log_spec.rb:53]: it's the same reason as below. The test was failing because `application_form.application_choices.first` was `nil`. Unless i am missing something?

File: app/views/candidate_mailer/application_submitted.text.erb (hunk 1 above)
Comment (avinhurry) [reply app/views/candidate_mailer/application_submitted.text.erb:25]: Changes made @EmmaFrith

File: app/views/candidate_mailer/application_submitted.text.erb (hunk 2 above)
Comment (avinhurry) [reply app/views/candidate_mailer/application_submitted.text.erb]: Changes made @EmmaFrith

File: app/views/candidate_mailer/application_submitted_apply_again.text.erb (hunk 3 above)
Comment (avinhurry) [reply app/views/candidate_mailer/application_submitted_apply_again.text.erb]: Changes made @EmmaFrith

File: app/views/candidate_mailer/application_submitted_apply_again.text.erb (hunk 4 above)
Comment (avinhurry) [reply app/views/candidate_mailer/application_submitted_apply_again.text.erb]: Changes made

File: app/views/candidate_mailer/application_submitted.text.erb [hunk 7]
```diff
@@ -1,13 +1,23 @@
...
+<% else %>
+
+  If your training provider decides to progress your application, they’ll contact you to organise an interview.
 
-<% @application_form.application_choices.each do |application_choice| %>
-* <%= application_choice.course_option.course.provider.name %> - <%= application_choice.course_option.course.name %> (<%= application_choice.course_option.course.code %>)
 <% end %>
 
-Sign in to your account anytime to check the progress of your application:
+<% unless @reject_by_default_date.nil? %>
```
[the remark this answers is not in the document]
Comment (avinhurry) [reply app/views/candidate_mailer/application_submitted.text.erb]: Ah yes, good spot, I guess I was being overly caution - i've removed.

File: spec/mailers/previews/candidate_mailer_preview.rb [hunk 8]
```diff
@@ -4,6 +4,7 @@ def application_submitted
       :completed_application_form,
       candidate: candidate,
       support_reference: 'ABCDEF',
+      application_choices: [FactoryBot.build_stubbed(:application_choice, :awaiting_provider_decision, course_option: course_option)],
```
Comment (despo) [inline spec/mailers/previews/candidate_mailer_preview.rb:7]:
```suggestion
      application_choices: FactoryBot.build_stubbed_list(:application_choice, 1, :awaiting_provider_decision, course_option: course_option),
```

File: app/views/candidate_mailer/application_submitted.text.erb (hunk 2 above)
Comment (EmmaFrith) [reply app/views/candidate_mailer/application_submitted.text.erb]: thanks @avinhurry!

File: app/views/candidate_mailer/application_submitted_apply_again.text.erb (hunk 3 above)
Comment (EmmaFrith) [reply app/views/candidate_mailer/application_submitted_apply_again.text.erb]: thanks @avinhurry!

File: app/views/candidate_mailer/application_submitted.text.erb (hunk 1 above)
Comment (EmmaFrith) [reply app/views/candidate_mailer/application_submitted.text.erb:25]: thanks @avinhurry!

**Comment by EmmaFrith** [comment]
Hey @avinhurry, when it's just one course, could we put the course details on the same line as the lead in sentence? Atm, there's a gap

Screenshot 2021-12-22 at 10 24 10

**Review by stevehook — APPROVED** [review]
:+1:
